File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var frequencySort = function(nums) {
    let freq = new Map();
    for(let i = 0; i < nums.length; i++){
        freq.set(nums[i], freq.get(nums[i]) + 1 || 1)
    }

    let freqArr = Array.from(freq)
    
    freqArr.sort((a,b) =>{
        if(a[1] !== b[1]){
            return a[1] - b[1]
        }
        return b[0] - a[0]
    })

    let res = [];
    
    for(let [key, val] of freqArr){
        for(let i = 0; i < val; i++){
            res.push(key)
        }
    }
    return res
};

File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.py
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        freq = Counter(nums)
        sorted_freq = sorted(freq.items(), key=lambda x: (x[1], -x[0]))
        res = []
        for key, val in sorted_freq:
            res.extend([key] * val)
        return res